research/src/pages/select/select.proposal.mdx

```diff
@@ -201,7 +202,47 @@ and is therefore considered to be invalid.
   - Presses the escape key
   - Selects one, or more (if `multiple` is true), by clicking or hitting `enter` or `space` key
 
-### Keyboard traversal
+### Typeahead
+
+#### The <a class="link" href="#typeahead-steps" id="typeahead-steps">typeahead steps</a> given a `<select>` and a `typed character` are as follows:
+
+1. Let `start new search` be false.
+1. If longer than the <a href="#typeahead-search-timeout">typeahead search timeout</a>
+   has elapsed since the previous invocation of the [typeahead steps](#typeahead-steps)
+   for `listbox`, then set `start new search` to true and set `buffer` to an empty string.
+1. Let `buffer` be `<select>`'s [`typeahead buffer`](#typeahead-buffer)
+1. Append `typed character` to `buffer`.
+1. Let `options` be the `<select>`'s list of options.
+1. Let `focusedIndex` be the index of `<select>`'s currently focused option.
+1. If `start new search` is true, then let `options for search` be the array of options given by
```

Comment:
This reorders the array for the search such that the items before the currently selected option (`Eu` in your example) come after the options following the currently selected one.  The focus can still jump backwards in the list to `Eu` (but if were other matches in the list after the currently selected option, it would jump to those first).

I think the way I tried to write this with JS functions is not at all clear though, so I've rewritten this using prose.  See changes here and in "find the first matching option".  Hopefully this is clearer now.

Of course I'm open to discussion on whether this is the final behavior we want to end up with.  At this point I'm just trying to match what I observe in Firefox/Chromium, which seems like reasonable behavior.  I've been using this to play with some of the interesting corner cases: https://jsfiddle.net/dandclark_msft/rx8tnps3/9/